Annotation: Add clean up job for old annotations (#26156)
grafana/grafana: docs/sources/administration/configuration.md

> **Note.** This setting has precedence over each individual rule frequency. If a rule frequency is lower than this value, then this value is enforced.

### max_annotation_age =

> Configures for how long alert annotations are stored. Default is 0, which keeps them forever.

Reviewer: For all max age settings I lack information about what kind of formats you could use to configure it. In docs and in .ini files.

Author: :+1:

### max_annotations_to_keep =

Configures max number of alert annotations that Grafana stores. Default value is 0, which keeps all alert annotations.

<hr>

## [annotations.dashboard]

> Dashboard annotations means that annotation are associated with the dashboard they are created on.

Reviewer 2: ```suggestion
Dashboard annotations means that annotations are associated with the dashboard they are created on.
```

> **Note.** This setting has precedence over each individual rule frequency. If a rule frequency is lower than this value, then this value is enforced.

> # Configures for how long alert annotations should be kept. Default is 0 which keeps them for ever

Reviewer 3: See default.ini edits.

Also, please correct the formatting from lines 979 to 985.

Reviewer: Done

> **Note.** This setting has precedence over each individual rule frequency. If a rule frequency is lower than this value, then this value is enforced.

### max_annotation_age =

Configures for how long alert annotations are stored. Default is 0, which keeps them forever.

> This setting should be expressed as an duration. Ex 6h (hours), 10d (days), 2w (weeks), 1M (month).

Reviewer 3: ```suggestion
This setting should be expressed as a duration. Examples: 6h (hours), 10d (days), 2w (weeks), 1M (month).
```

### max_annotations_to_keep =

Configures max number of alert annotations that Grafana stores. Default value is 0, which keeps all alert annotations.

<hr>

## [annotations.dashboard]

Dashboard annotations means that annotations are associated with the dashboard they are created on.

### max_age

Configures how long dashboard annotations are stored. Default is 0, which keeps them forever.

> This setting should be expressed as an duration. Ex 6h (hours), 10d (days), 2w (weeks), 1M (month).

Reviewer 3: ```suggestion
This setting should be expressed as a duration. Examples: 6h (hours), 10d (days), 2w (weeks), 1M (month).
```

### max_annotations_to_keep

Configures max number of dashboard annotations that Grafana stores. Default value is 0, which keeps all dashboard annotations.

## [annotations.api]

API annotations means that the annotations have been created using the API without any association with a dashboard.

### max_age

Configures how long Grafana stores API annotations. Default is 0, which keeps them forever.

> This setting should be expressed as an duration. Ex 6h (hours), 10d (days), 2w (weeks), 1M (month).

Reviewer 3: ```suggestion
This setting should be expressed as a duration. Examples: 6h (hours), 10d (days), 2w (weeks), 1M (month).
```
